rbc/external.py

```diff
@@ -8,52 +8,67 @@
 
 class External:
     @classmethod
-    def fromobject(cls, *args, typing=True, lowering=True):
+    def declare(cls, *args):
+        return cls.external(*args, register=False)
+
+    @classmethod
+    def external(cls, *args, register=True):
         """
         Parameters
         ----------
         signature : object (str, ctypes function, python callable, numba function)
             Any object convertible to a Numba function via Type.fromobject(...).tonumba()
-        typing : bool
-            Indicates if External should do typing or not. Default is True
-        lowering: bool
-            Indicates if External should do lowering or not. Default is True. Except for
-            very specific cases, typing and lowering should be True
+        register : bool
+            Indicates if External should do typing/lowering or not. Default is True
         """
-        # Make inner function for the actual work
+        try:
+            targetinfo = TargetInfo()
+        except RuntimeError:
+            targetinfo = TargetInfo.dummy()
+
+        # If we already have a targetinfo registered, then, use an empty context manager
+        try:
+            with targetinfo:
+                pass
+        except AssertionError:
+            import contextlib
+            targetinfo = contextlib.nullcontext()
+
```

Comment:
This part of the code is a bit obscure. I wonder if there's a better way to know if a `TargetInfo` object already exists and if it is being used.

We have a few corner cases here:
1. When registering a new registry inside `irtools.py`, a target info was already registered and it is being used
2. When one imports `from rbc.externals import cmath` for instance, a target info probably do not exists

Reply:
The fix is here:
https://github.com/xnd-project/rbc/pull/292/commits/3850649f980b8517efefa930a55d4c06a3e2a5bd
that optionally allows nesting target info contexts.

Reply:
See #299 .